log/logger.ts

```diff
@@ -56,22 +57,40 @@ export class Logger {
     options: LoggerOptions = {},
   ) {
     this.#loggerName = loggerName;
-    this.#level = getLevelByName(levelName);
+    /* TODO: Remove this unnecessary typecast after 0.211.0 */
+    this.#level = getLevelByName(levelName) as LogLevel;
     this.#handlers = options.handlers || [];
   }
 
-  get level(): LogLevels {
+  /**
+   * Use this to retrieve the current numeric log level.
+   *
+   * @returns {number} - Deprecated (will return {@linkcode LogLevel} after 0.211.0)
```

Comment:
There's no need to define types in JSDocs as the function itself already does that. Ditto to other instances in the PR.

Reply:
Was my thinking as well, but how do you put out deprecation notices on certain parts of the signature otherwise?

Using a `@deprecated` tag marks the whole function as deprectated. To me that seemed controdictionary. But if that's how you want to roll here, I am down to change it.